### ISSUE TITLE
Query regarding when to use bias in affine transform

### ISSUE THREAD

**Issue by ratishsp** [OpenNMT/OpenNMT-py#624]
Hi,
In CopyGenerator the linear transform uses bias: `self.linear = nn.Linear(input_size, len(tgt_dict))`
In GlobalAttention, the linear transform does not use bias `self.linear_in = nn.Linear(dim, dim, bias=False)`. Is there any guideline on when to use bias and when to not? Or is it verified empirically?
Thanks

**Comment by sebastianGehrmann** [comment]
In the implementations, we followed standard practices from published papers. For example, we roughly implemented the copy version here: https://arxiv.org/pdf/1704.04368.pdf 

I did some minor empirical testing some time ago and it seems that the current biases are helping the model.
